### DISCUSSION TITLE
How to tell one manager to the use the version of another manager? (helmv3 -> asdf)

### DISCUSSION THREAD

**Discussion by julianxhokaxhiu** [renovatebot/renovate#19929] [Request Help]
Hi,

I'm trying to set up the renovate bot to accomplish a straightforward task: update one package version using another.

The full explanation of this issue can be found here: https://github.com/nearform/k8s-kurated-addons/issues/24

Long story short: I have a Helm Chart ( ArgoCD ) which has its version that is NOT the same as ArgoCD, but inside of it there is defined the ArgoCD version being used. This version has to be now used to bump the asdf `.tool-version` file where the ArgoCD line exists.

Is it possible to accomplish such a thing, and if so how? I looked through the documentation for some events callbacks or similar but I found nothing. I'm currently running the self-hosted https://github.com/[renovatebot/github-action](https://github.com/renovatebot/github-action) as you can see here: https://github.com/nearform/k8s-kurated-addons/blob/main/.github/workflows/renovate.yaml#L23-L29

Thank you in advance for any help.

**Comment by rarkins** [comment]
The description here doesn't quite seem the same as in the other issue.

First question: are these two dependencies already getting updated, just not together?

If so then what are the branch names for the two PRs?

**Reply by julianxhokaxhiu** [reply]
> First question: are these two dependencies already getting updated, just not together?

Well indeed what I am asking here is step 1, step 2 would then merge these two actions into one single PR. They are two dependencies that might get updated, but separately. One will be managed via helmv3 and the other via asdf. The problem is they are not always pointing to the same ArgoCD version.

> If so then what are the branch names for the two PRs?

None atm. I'm just asking if it is possible to do this using renovate bot. I'm basically tasting the waters :)

**Reply by rarkins** [reply]
Unfortunately I'm not able to follow enough of what you've described in order to help. I think the best thing would be for you to set up a minimal reproduction repo with the two dependencies to demonstrate what you mean.

**Reply by julianxhokaxhiu** [reply]
Fair :) Let me setup a simple repo so you can play with it. Thanks!

**Reply by julianxhokaxhiu** [reply]
Allright, as I was preparing the test repo you can find here ( https://github.com/julianxhokaxhiu/renovate-test ) I noticed that renovate does not support the `argocd` plugin on the asdf manager. So I opened an issue ( https://github.com/renovatebot/renovate/issues/19940 ) and a PR ( https://github.com/renovatebot/renovate/pull/19941 ). Once those will both be solved we can move on with this discussion here as a second PR should come up that should bump the `.tool-versions`. Once those two PRs will be ready we can see how to make them one and especially, how to tell the `asdf` manager that the version it has to use is another one and not the one from the plugin itself.

**Reply by julianxhokaxhiu** [reply]
So the PR #19941 has been merged now but I noticed the renovatebot is still not picking up the `argocd` package, see https://github.com/renovatebot/renovate/pull/19941#issuecomment-1399224061

I did now add a unit test coverage for this case ( see https://github.com/renovatebot/renovate/pull/19983 ) and even though it passes ( which means the extraction is working ) I'm still puzzled because I'm not clear why the renovatebot is not able to see the package when being run.

Any ideas? /cc @viceice

**Reply by rarkins** [reply]
Is it possible to have a reproduction repo with just that dependency, or at least only asdf dependencies?

**Reply by julianxhokaxhiu** [reply]
It's already there: https://github.com/julianxhokaxhiu/renovate-test/blob/master/.tool-versions my repo is not seeing any PR coming, reason is in https://github.com/renovatebot/renovate/pull/19941#issuecomment-1399224061

**Reply by rarkins** [reply]
It looks fine to me:

```
DEBUG: packageFiles with updates (repository=julianxhokaxhiu/renovate-test, baseBranch=master)
       "config": {
         "asdf": [
           {
             "packageFile": ".tool-versions",
             "deps": [
               {
                 "currentValue": "2.5.4",
                 "depName": "argocd",
                 "datasource": "github-releases",
                 "packageName": "argoproj/argo-cd",
                 "versioning": "semver",
                 "extractVersion": "^v(?<version>\\S+)",
                 "depIndex": 0,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "2.5.7",
                     "newValue": "2.5.7",
                     "releaseTimestamp": "2023-01-18T03:00:02.000Z",
                     "newMajor": 2,
                     "newMinor": 5,
                     "updateType": "patch",
                     "branchName": "renovate/argocd-2.x"
                   }
                 ],
                 "warnings": [],
                 "sourceUrl": "https://github.com/argoproj/argo-cd",
                 "registryUrl": "https://github.com",
                 "currentVersion": "2.5.4",
                 "isSingleVersion": true,
                 "fixedVersion": "2.5.4"
               }
             ]
           }
         ],
```

**Reply by rarkins** [reply]
@julianxhokaxhiu just make a commit to the master branch so that Renovate rescans its dependencies

**Reply by julianxhokaxhiu** [reply]
Ahhh! Ok so each run on https://app.renovatebot.com/dashboard#github/julianxhokaxhiu/renovate-test unless I commit, is basically cached? Thanks, I'll do it now

//EDIT: Nvm it seems it is getting it right now, not sure what happened but I didn't touch anything. I can see the dictionary on https://app.renovatebot.com/dashboard#github/julianxhokaxhiu/renovate-test/985476335 good!

**Reply by julianxhokaxhiu** [reply]
So getting back to this discussion now, I have both the argocd PR ( https://github.com/julianxhokaxhiu/renovate-test/pull/4 ) and the helm chart bump PR ( https://github.com/julianxhokaxhiu/renovate-test/pull/2 ). As you can notice, the latter has a different versioning BUT inside the chart, there is a property ( https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml#L2 ) stating which argocd version it is using. How can I transfer this info to the asdf manager now for the `argocd` plugin?

In other words, how can I use as a version to compare the current one defined in `.tool-versions` with the one defined here ( https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml#L2 )?

Thank you in advance

**Reply by rarkins** [reply]
One of those dependencies has the name "argocd" and the others has a hyphen "argo-cd". Easiest would be to rename one of those in the next major release so that they're always grouped

**Reply by julianxhokaxhiu** [reply]
Right, but that would bump the two components into one PR. That solves partially the problem. The version of `argocd` in `.tool-versions` will not be in sync with the chart [defined one](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml#L2). How can I tell to the renovate bot that instead of using the "resolved one" to use [this one](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml#L2)?

**Reply by rarkins** [reply]
Can you create a manual PR in this same repo showing the exact change you'd want, and briefly describing the logic behind that?

**Reply by julianxhokaxhiu** [reply]
@rarkins there you go: https://github.com/julianxhokaxhiu/renovate-test/pull/5

**Reply by rarkins** [reply]
Thanks, I think I finally understand what you mean. Unfortunately, I can't think of a way to achieve it. The first step would be for the helm datasource to return semantic information about appVersion + "what app is that version referring to". The problem is that there's no canonical way to associate that with the same argocd you mean in asdf. Even if there were, the next step is working out how exactly Renovate should "align" the two in the same PR. Essentially this is some form of "peer dependency" concept in helm, but not formalized.

**Reply by julianxhokaxhiu** [reply]
Right, exactly. I was wondering if it was possible to accomplish this in the first place as I wasn't able to spot the functionality through documentation.

Is there any hope you'll ever implement inter-dependency between managers in case? I noticed that asdf has some concept of datasource, maybe we could add "manager" or something similar that would tell how the resolution should work.

If not, it's fine anyway. I'll find another way to accomplish the same thing :)

**Reply by rarkins** [reply]
I don't think this is feasible because there isn't any semantic data to link the two of them in the first place.

**Reply by julianxhokaxhiu** [reply]
I see. Well it was worth a shot. I'll find another way then. Thanks a lot!

**Reply by viceice** [reply]
probably easiest solution it to validate with CI check, so it at least stops from merging. 🤔
